Marlin/src/HAL/HAL_LPC1768/HardwareSerial.cpp

```diff
@@ -35,301 +35,606 @@ volatile uint32_t UART0RxQueueWritePos = 0, UART1RxQueueWritePos = 0, UART2RxQue
 
-void HardwareSerial::begin(uint32_t baudrate) {
-  uint32_t Fdiv, pclkdiv, pclk;
-
-  if (PortNum == 0) {
-    LPC_PINCON->PINSEL0 &= ~0x000000F0;
-    LPC_PINCON->PINSEL0 |= 0x00000050;  /* RxD0 is P0.3 and TxD0 is P0.2 */
-    /* By default, the PCLKSELx value is zero, thus, the PCLK for
-       all the peripherals is 1/4 of the SystemFrequency. */
-    /* Bit 6~7 is for UART0 */
-    pclkdiv = (LPC_SC->PCLKSEL0 >> 6) & 0x03;
-    switch (pclkdiv) {
-      case 0x00:
-      default:
-        pclk = SystemCoreClock / 4;
-        break;
-      case 0x01:
-        pclk = SystemCoreClock;
-        break;
-      case 0x02:
-        pclk = SystemCoreClock / 2;
-        break;
-      case 0x03:
-        pclk = SystemCoreClock / 8;
-        break;
-    }
+  void HardwareSerial::begin(uint32_t baudrate) {
+    uint32_t Fdiv;
+     uint32_t pclkdiv, pclk;
 
-    LPC_UART0->LCR = 0x83;       /* 8 bits, no Parity, 1 Stop bit */
-    Fdiv = ( pclk / 16 ) / baudrate ;   /*baud rate */
-    LPC_UART0->DLM = Fdiv / 256;
-    LPC_UART0->DLL = Fdiv % 256;
-    LPC_UART0->LCR = 0x03;      /* DLAB = 0 */
-    LPC_UART0->FCR = 0x07;       /* Enable and reset TX and RX FIFO. */
+     if ( PortNum == 0 )
+     {
+   	LPC_PINCON->PINSEL0 &= ~0x000000F0;
+   	LPC_PINCON->PINSEL0 |= 0x00000050;  /* RxD0 is P0.3 and TxD0 is P0.2 */
+   	/* By default, the PCLKSELx value is zero, thus, the PCLK for
+   	all the peripherals is 1/4 of the SystemFrequency. */
+   	/* Bit 6~7 is for UART0 */
+   	pclkdiv = (LPC_SC->PCLKSEL0 >> 6) & 0x03;
+   	switch ( pclkdiv )
+   	{
+   	  case 0x00:
+   	  default:
+   		pclk = SystemCoreClock/4;
+   		break;
+   	  case 0x01:
+   		pclk = SystemCoreClock;
+   		break;
+   	  case 0x02:
+   		pclk = SystemCoreClock/2;
+   		break;
+   	  case 0x03:
+   		pclk = SystemCoreClock/8;
+   		break;
+   	}
 
-    NVIC_EnableIRQ(UART0_IRQn);
+       LPC_UART0->LCR = 0x83;		/* 8 bits, no Parity, 1 Stop bit */
+   	Fdiv = ( pclk / 16 ) / baudrate ;	/*baud rate */
+       LPC_UART0->DLM = Fdiv / 256;
+       LPC_UART0->DLL = Fdiv % 256;
+   	LPC_UART0->LCR = 0x03;		/* DLAB = 0 */
+       LPC_UART0->FCR = 0x07;		/* Enable and reset TX and RX FIFO. */
 
-    LPC_UART0->IER = IER_RBR | IER_THRE | IER_RLS;   /* Enable UART0 interrupt */
-  }
-  else if (PortNum == 1) {
-    LPC_PINCON->PINSEL4 &= ~0x0000000F;
-    LPC_PINCON->PINSEL4 |= 0x0000000A;  /* Enable RxD1 P2.1, TxD1 P2.0 */
-
-    /* By default, the PCLKSELx value is zero, thus, the PCLK for
-    all the peripherals is 1/4 of the SystemFrequency. */
-    /* Bit 8,9 are for UART1 */
-    pclkdiv = (LPC_SC->PCLKSEL0 >> 8) & 0x03;
-    switch (pclkdiv) {
-      case 0x00:
-      default:
-        pclk = SystemCoreClock / 4;
-        break;
-      case 0x01:
-        pclk = SystemCoreClock;
-        break;
-      case 0x02:
-        pclk = SystemCoreClock / 2;
-        break;
-      case 0x03:
-        pclk = SystemCoreClock / 8;
-        break;
-    }
+      	NVIC_EnableIRQ(UART0_IRQn);
```

Comment:
Is the priority ever set for these interrupts, cant have serial interrupting things? priority 5, same as usb will be fine I think` NVIC_SetPriority(USB_IRQn, NVIC_EncodePriority(0, 5, 0));`